src/main/java/com/capgemini/edem/dgs/datafetchers/ProductDatafetcher.java

```diff
@@ -20,15 +22,32 @@ public class ProductDatafetcher {
 
   @Autowired
   private ProductRepository productRepository;
+  private Counter testCounter;
+  private Counter testCounterError;
+  private Counter testCounterAll;
+  private Counter testCounterAllError;
+  
+  public ProductDatafetcher(MeterRegistry meterRegistry) {
+	  testCounter = meterRegistry.counter("get_Product_id");
+	  testCounterError = meterRegistry.counter("get_Product_id_error");
+	  testCounterAll = meterRegistry.counter("get_Products_all");
+	  testCounterAllError = meterRegistry.counter("get_Products_all_error");
+}
 
   @DgsQuery(field = QUERY.Product)
   public Mono<? extends IProduct> getProduct(@InputArgument String id) {
-    return this.productRepository.findById(id);
+	  testCounter.increment();
```

Comment:
@aalamino , please try to add instrumentation without modifying functional code. See more info about that [here](https://netflix.github.io/dgs/advanced/instrumentation/)

Reply:
ok, added custom instrumentation without modifying functional code